### ISSUE TITLE
Wierd bug with the monitor renderer and unable to change it

### ISSUE THREAD

**Issue by EmmaKnijn** [Shrimpstress/cc-tweaked-fabric#17]
So I have this problem that I'm unable to change the monitor renderer, in forge I would've just gone to mod list and changed it there, but i can't see an option to change it. Even in the `computercraft.json` file it doesn't show anything about the monitor renderer. The reason I want to change it is because I have this wierd bug with monitors
( https://cdn.discordapp.com/attachments/764229164982861847/767768480711704576/unknown.png )
Text does display but anything that isn't text just shows this character.
I'm running 1.16.3 on a server.
You can try and replicate it by buying a pentium g2020 computer with internal graphics, since this is the only pc that it happens on on the server I play on, the easiest way to fix this is make a way to change monitor renderer. Because that worked on forge, I hope you can help me with this.

**Comment by Ueaj-Kerman** [comment]
not sure if this was fixed in a recent version, we're kinda behind upstream rn

**Comment by SquidDev** [comment]
The bug has never been fixed upstream, as we've never been able to work out quite what's going on. 

That said [the config][config] is definitely missing a few options (including `monitor_renderer`), so probably worth syncing that up.

[config]: https://github.com/Zundrel/cc-tweaked-fabric/blob/fabric/src/main/java/dan200/computercraft/shared/util/Config.java

**Comment by SquidDev** [comment]
@RubenHetKonijn Just a couple of things to check:

 - Could you take a screenshot of the whole debug screen (F3)? Just so I can see OpenGL information.
 - Are you running with any resource packs?

**Comment by EmmaKnijn** [comment]
Here you go: 
![2020-10-21_09 40 34](https://user-images.githubusercontent.com/64010592/96688750-d1699a80-1381-11eb-929f-25bdc97f4ade.png)
![2020-10-21_09 40 45](https://user-images.githubusercontent.com/64010592/96688756-d29ac780-1381-11eb-88a4-36a84530db6f.png)

If you need the mods list: 
![MultiMC_xLoLqJdSkx](https://user-images.githubusercontent.com/64010592/96688820-e1817a00-1381-11eb-93a5-ea7f540b91f6.png)

I don't think sodium has to do with the render issue, since the same happens on the forge version with and without optifine, and same happens on fabric without sodium

**Comment by SquidDev** [comment]
Fantastic, that's very helpful! I can see what's going on, just not sure why.

Hopefully just one last thing: would you be able to build a large monitor (8x6 blocks) in a single player world, and run the following program on an adjacent computer: `wget run https://gist.githubusercontent.com/SquidDev/b6e144235147c59667dbc53fef1ac4ed/raw/dump-chars.lua`, then take a screenshot of the monitor.

It appears that the space character is being mapped to the character before it, which makes no sense given other characters (such as letters) don't appear to have this issue. That program just dumps all characters, so should allow me to see if there's any others being misplaced.

**Comment by EmmaKnijn** [comment]
![2020-10-21_10 41 05](https://user-images.githubusercontent.com/64010592/96695430-f06c2a80-1389-11eb-8cac-c458dcdb9934.png)

Here you go :)

**Comment by SquidDev** [comment]
@Devan-Kerman et all: Just as a heads up, the issue has been fixed upstream (:heart: Lemmmy, but oh I hate rendering issues).

That said, probably worth adding the config option. I've no clue how well the TBO renderer will play with something like Canvas, so probably worth allowing people to disable it.

**Comment by EmmaKnijn** [comment]
Yes, I hope it gets into fabric because that is what I mainly play at the moment, do you have an ETA for that?

**Comment by Ueaj-Kerman** [comment]
soon:tm:, hopefully I'll catch everything up with upstream on the weekend, I got school rn

**Comment by EmmaKnijn** [comment]
The issue has been fixed upstream: https://github.com/SquidDev-CC/CC-Tweaked/releases/tag/v1.15.2-1.93.1
